File: Frontend/src/app/modules/experience/models/experience-view-model.ts
export class ExperienceViewModel {
  id!: number;
  creationDate!: string;
  firstModificationDate!: string | null;
  lastModificationDate!: string | null;
  createdByUserId!: number | null;
  firstModifiedByUserId!: number | null;
  lastModifiedByUserId!: number | null;
  isDeleted!: boolean;
  deletionDate!: string | null;
  deletedByUserId!: number | null;
  mustDeletedPhysical!: boolean | null;
  nameEn!: string;
  nameAr!: string;

  parentId!: number | null;
  parentName!: string;

}

File: Backend/TemplateService.Core/Models/ViewModels/LookUps/Experience/ExperienceViewModel.cs
#region Using ...
using System;
using System.Diagnostics;
#endregion

/*


*/
namespace TemplateService.Core.Models.ViewModels
{
    /// <summary>
    /// 
    /// </summary>
    [DebuggerDisplay("Id={Id}, CreationDate={CreationDate}, Content={Content}, UserId={UserId}, IsPostive={IsPostive}")]
    public class ExperienceViewModel : Base.BaseViewModel
    {
        #region Constructors
        /// <summary>
        /// Initializes a new instance from type
        /// ExperienceViewModel
        /// </summary>
        public ExperienceViewModel()
        {
        }
        #endregion

        #region Properties

        #region IEntityIdentity
        public long Id { get; set; }
        #endregion

        #region IDateTimeSignature
        public DateTime CreationDate { get; set; }
        public DateTime? FirstModificationDate { get; set; }
        public DateTime? LastModificationDate { get; set; }
        #endregion

        #region IEntityUserSignature
        public long? CreatedByUserId { get; set; }
        public long? FirstModifiedByUserId { get; set; }
        public long? LastModifiedByUserId { get; set; }
        #endregion

        #region IDeletionSignature
        public bool IsDeleted { get; set; }
        public DateTime? DeletionDate { get; set; }
        public long? DeletedByUserId { get; set; }
        public bool? MustDeletedPhysical { get; set; }
        #endregion

        public string NameEn { get; set; }
        public string NameAr { get; set; }
        public string ParentName { get; set; }
        public long? ParentId { get; set; }


        #endregion
    }
}
